core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -247,7 +249,7 @@ class KafkaApis(val requestChannel: RequestChannel,
         case ApiKeys.ALTER_PARTITION_REASSIGNMENTS => maybeForward(request, handleAlterPartitionReassignmentsRequest)
         case ApiKeys.LIST_PARTITION_REASSIGNMENTS => handleListPartitionReassignmentsRequest(request)
         case ApiKeys.OFFSET_DELETE => handleOffsetDeleteRequest(request)
-        case ApiKeys.DESCRIBE_CLIENT_QUOTAS => handleDescribeClientQuotasRequest(request)
+        case ApiKeys.DESCRIBE_CLIENT_QUOTAS => maybeForward(request, handleDescribeClientQuotasRequest)
```

Comment:
So, the pattern we've been following so far is to handle "read" operations locally on the broker, and only forward "write" operations to the controller.  We should probably be consistent about this

Reply:
For reads, should we return the latest state of the quotas as far as the metadata records are concerned, or should we return the state as known to QuotaManagers?

If it's the former, seems like this is duplicating efforts of the quorum controller (processing records to build state). Even if it's the latter, it seems like there could be some consistency issues. 

If a DescribeClientQuota request is made following an AlterClientQuota request, and the broker hasn't replicated and processed the QuotaRecord, we might violate read-after-write semantics. Another issue is that this request could return different results depending on which broker handled it, since there can be differences in metadata propagation time. 

Maybe not a big deal for this API, but worth noting. Thoughts?

Reply:
Yes, there are consistency issues.  This is an existing issue in Kafka: we do not guarantee read-after-write consistency with administrative operations.  This is the case for things like creating a topic, where the topic probably won't show up immediately in the next listTopics, but you have to keep polling for it.  I do think we should fix this, but not by redirecting metadata read operations to the controller... that doesn't scale.

Let's follow the existing pattern for now and think about how to improve this in a follow-on KIP.  Now that we have a monotonic metadata offset index, there are some obvious options that suggest themselves.  But we have to make sure what we do for metadata fits in with global kafka.